include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -176,20 +207,20 @@ struct erf_helper<FloatingPoint NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScala
 {
 	static FloatingPoint __call(NBL_CONST_REF_ARG(FloatingPoint) _x)
 	{
-		const FloatingPoint a1 = 0.254829592;
-		const FloatingPoint a2 = -0.284496736;
-		const FloatingPoint a3 = 1.421413741;
-		const FloatingPoint a4 = -1.453152027;
-		const FloatingPoint a5 = 1.061405429;
-		const FloatingPoint p = 0.3275911;
-
-		FloatingPoint sign = sign(_x);
+		const FloatingPoint a1 = FloatingPoint(0.254829592);
+		const FloatingPoint a2 = FloatingPoint(-0.284496736);
+		const FloatingPoint a3 = FloatingPoint(1.421413741);
+		const FloatingPoint a4 = FloatingPoint(-1.453152027);
+		const FloatingPoint a5 = FloatingPoint(1.061405429);
+		const FloatingPoint p = FloatingPoint(0.3275911);
+
+		FloatingPoint _sign = FloatingPoint(sign(_x));
```

Comment:
no just a C-cast will not work

You need the `NBL_FLOAT64_LITERAL` macro because it adds a platform dependant suffix

> So a double bag of `float_t(NBL_FLOAT64_LITERAL(0.3232983928))` is needed

The reason is that the constant literal itself is not fp32, so its already at reduced/truncated precision by the time you C-cast it to float64 (promoting it)

You still have the old problem https://github.com/Devsh-Graphics-Programming/Nabla/pull/821#discussion_r1947978451

p.S. maaaybe because you're assigning straight to `const FloatingPoint` the `NBL_FLOAT64_LITERAL` macro is enough.